File: src/0003_longest_substring_without_repeating_characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> seen;
        int maxLength = 0;
        int left = 0;
        for (int right=0; right<s.length(); right++){
            while (seen.count(s[right]) > 0) {
                seen.erase(s[left]);
                left++;
            }
            seen.insert(s[right]);
            maxLength = max(right - left + 1, maxLength);
        }
        return maxLength;
    }
};

File: src/0003_longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left=0
        maxlen = 0
        seen = set()
        for right in range(len(s)):
            while s[right] in seen:
                seen.remove(s[left])
                left += 1

            seen.add(s[right])
            maxlen = max(right - left + 1, maxlen)

        return maxlen
